sipXyealink/src/org/sipfoundry/sipxconfig/phone/yealink/YealinkPhone.java

```diff
@@ -215,6 +223,18 @@ public boolean getHasHD() {
     public String getDirectedCallPickupString() {
         return getPhoneContext().getPhoneDefaults().getDirectedCallPickupCode();
     }
+    
+    public String getPhonebookType(){
+        //Setting setting = getSettings().getSetting(PHONEBOOKTYPE);
+        Setting settings = getSettings();
+        Setting setting = settings.getSetting(PHONEBOOKTYPE);
+        String settingValue = (null == setting ? null : (setting.getValue() == null ? null : setting
+                        .getValue()));
+        if (!StringUtils.isEmpty(settingValue)) {
+                    return settingValue;
+                }
+        return "2";
```

Comment:
pre commit not work:
build]$ make precommit
mak/05-build.mk:79: target 'sipXyealink.changes' given more than once in the same rule
mak/05-build.mk:84: target 'sipXyealink.if-changed' given more than once in the same rule
mak/05-build.mk:106: target 'sipXyealink.dist' given more than once in the same rule
mak/05-build.mk:112: target 'sipXyealink.install' given more than once in the same rule
mak/05-build.mk:116: target 'sipXyealink.clean' given more than once in the same rule
mak/05-build.mk:121: target 'sipXyealink.check' given more than once in the same rule
mak/05-build.mk:126: target 'sipXyealink.all' given more than once in the same rule
mak/10-rpm.mk:88: target 'sipXyealink.rpm.changes' given more than once in the same rule
mak/10-rpm.mk:93: target 'sipXyealink.rpm.if-changed' given more than once in the same rule
mak/10-rpm.mk:111: target 'sipXyealink.srpm' given more than once in the same rule
mak/20-list-dependencies.mk:83: target 'sipXyealink.deps-missing' given more than once in the same rule
mak/20-list-dependencies.mk:89: target 'sipXyealink.deps' given more than once in the same rule
make: *** No rule to make target 'precommit'.  Stop.